src/providers/ad/ad_subdomains.c

```diff
@@ -1646,6 +1646,13 @@ static void ad_check_root_domain_done(struct tevent_req *subreq)
         goto done;
     }
 
+    ret = sysdb_attrs_add_string(state->reply[0], AD_AT_DOMAIN_NAME,
+                                 state->forest);
```

Comment:
I'm confused here: why `AD_AT_TRUST_PARTNER` ("trustPartner") == `AD_AT_DOMAIN_NAME` ("cn")?

`trustPartner` - "The name of the domain with which a trust exists."
If this object represents trusted domain, why its `trustPartner` points to itself?

Reply:
It also strikes me that the same information is stored in two attributes that do not seem to be closely related to each other.

Reply:
If we are querying a child domain for forest root information, then perhaps the trustPartner of the child domain Trusted Domain Object is the forest root, which is also the "cn" attribute that

**During this processing it is expected that the 'cn' attribute is set and contains the name of the forest root.**

Reply:
@sumit-bose ping